src/sql/workbench/contrib/notebook/browser/notebookActions.ts

```diff
@@ -213,12 +219,14 @@ export class RunAllCellsAction extends Action {
 	constructor(
 		id: string, label: string, cssClass: string,
 		@INotificationService private notificationService: INotificationService,
-		@INotebookService private _notebookService: INotebookService
+		@INotebookService private _notebookService: INotebookService,
+		@IAdsTelemetryService private _telemetryService?: IAdsTelemetryService,
 	) {
 		super(id, label, cssClass);
 	}
 	public async run(context: URI): Promise<boolean> {
 		try {
+			this._telemetryService?.sendActionEvent(TelemetryKeys.TelemetryView.Notebook, TelemetryKeys.NbTelemetryAction.RunAll);
```

Comment:
You can remove the ? from all these now

Reply:
What if the _telemetryService parameter is passed as undefined (as I did in some tests). I think it's foolproof to have that.

Reply:
It should never be - for tests use https://github.com/microsoft/azuredatastudio/blob/main/src/sql/platform/telemetry/common/adsTelemetryService.ts#L217